### ISSUE TITLE
Requests for admin panel and admin controls

### ISSUE THREAD

**Issue by ryooan** [Metaculus/metaculus#1090]
- [x] 1. There should be a toggle in admin for projects to show in main feed, currently you can only change this from the dropdown on the front end.
- [x] 2. A dedicated place to edit the hot topics and add and remove questions from them, instead of having to go into each post and add the topic project to it.
- [x] 3.  An admin-only button to ban users from the front end. Ideal solution would have the following: Every profile (and/or every comment?) has one admin-only menu option on the front end that with one click deactivates a user and deletes every comment they've made (and probably every question/post they've made too). Should probably pop up an "are you sure". But this would greatly speed up deleting the spam comments we get (seems like we get a couple a day on average, and probably only going to get worse).
- [x] 4. On posts, what's the difference between "default project" and "Projects"? Are both needed? If so, some explanation for what they mean and how to use them would be appreciated.
- [x] 5.  The questions and posts panel should both let you filter by project.
- [x] 6.  The "questions" panel could use a set of more informative columns, maybe like the "posts" panel has, see picture below.
 
![image](https://github.com/user-attachments/assets/f801eb84-9f87-49df-88ba-0d910a930a6a)

- [x] 7. Ideally as well would be great if instead of just prepending the question type ("numeric", etc) to the name of the question that was in its own column and filterable. Would be ideal in the posts view too if possible and if it makes sense to do that.
- [x] 8.  Posts have a link down to the question, which is very helpful, but would be great if questions had a link up to the post they live in.
- [ ] 9.  It's my understanding that the dates and times shown on the post are just copied from the question level, and should be edited at the question level. If that's correct, is there any reason these can't be linked both directions so that we can just edit them at the post level if we want? (And if that's possible it would be great to include CP reveal time in the post fields).
- [x] 10.  Group questions are hard to navigate in admin. There are posts, which link to a group of questions within them. But the group of questions page has little to no information at all. And then there are separate "questions" objects for each subquestion. Can the "group of questions" object contain a list of all of the individual questions with a link to edit them?
- [x] 11.  There's no search field in admin for Notebooks. This wouldn't be a big issue but for the spam notebooks, which unfortunately due to their volume necessitate paging through to find content (requested by Christian)
- [x] 12. Give bulletins names so they don't show up as "bulletin object (x)", where x is the number of the bulletin. https://www.metaculus.com/admin/misc/bulletin/
- [ ] 13. (Nice to have, not strictly necessary): A bulletin preview for admins, so you can see what a bulletin will look like on the site before it goes live.

**Comment by ryooan** [comment]
This is all I've run across for the time being, removing myself as assignee to unblock it, probably not high priority but there's enough here that would be valuable for someone to tackle when they have the time.

**Comment by lsabor** [comment]
1. Addressed
2. It's not elegant, but it is addressed: metaculus.com/admin/projects/project/15854/change/ there are now attached all relevant questions at the bottom which can be removed via checkmark. To add new ones, there is a link above the list area that says "Add Posts" leading to a search page which adds them
3. Not addressed - still needs to be done
4. @hlbmtc can you answer this?
5. addressed
6. addressed - feel free to request tweaks
7. addressed
8. addressed
9. too complicated at the moment. all times relavant to scoring should go on the question, only times relevant to publication should go on the post. This needs work
10. not addressed - still needs to be done
11. addressed

Changes will be visible at next deployment

**Comment by lsabor** [comment]
For ease of reference. What's still needed is:
3. An admin-only button to ban users from the front end. Ideal solution would have the following:
* Every profile (and/or every comment?) has one admin-only menu option on the front end that with one click deactivates a user and deletes every comment they've made (and probably every question/post they've made too). Should probably pop up an "are you sure". But this would greatly speed up deleting the spam comments we get (seems like we get a couple a day on average, and probably only going to get worse).
4. On posts, what's the difference between "default project" and "Projects"? Are both needed? If so, some explanation for what they mean and how to use them would be appreciated.
9. It's my understanding that the dates and times shown on the post are just copied from the question level, and should be edited at the question level. If that's correct, is there any reason these can't be linked both directions so that we can just edit them at the post level if we want? (And if that's possible it would be great to include CP reveal time in the post fields).
10. Group questions are hard to navigate in admin. There are posts, which link to a group of questions within them. But the group of questions page has little to no information at all. And then there are separate "questions" objects for each subquestion. Can one or both of the "post" or "group of questions" object contain a list of all of the individual questions with a link to edit them?

**Comment by lsabor** [comment]
Addressed number 3

**Comment by ryooan** [comment]
Awesome, thank you so much @lsabor ! Will take a look at these this week and follow up.

On #3, that's super useful and I love it. As a follow up to that, could we make it so that deactivated users profiles are shown as deactivated? For example, I used soft delete on this user but their spam account is still visible and looks no different from normal if you can get to the profile. https://www.metaculus.com/accounts/profile/227321/?mode=overview

I think ideal behavior would be something like showing none of the normal user page and instead just text like "[Username]'s account has been deactivated"

**Comment by ryooan** [comment]
Another relevant one I just thought of, ability to filter by username on the comments feed. Also would be nice to adjust column widths. https://www.metaculus.com/admin/comments/comment/

@lsabor

**Comment by lsabor** [comment]
>Another relevant one I just thought of, ability to filter by username on the comments feed.

Good idea, I made [a separate ticket](https://github.com/Metaculus/metaculus/issues/1580) though for that feature.

**Comment by lsabor** [comment]
>For example, I used soft delete on this user but their spam account is still visible and looks no different from normal

Oh you're totally right. That should just go straight to a 404, whoops.
Easy fix

**Comment by ryooan** [comment]
Adding one more about bulletins, currently bulletin names are "bulletin object (x)" where x is the number of bulletin created. Would be great to either be able to name those, or grab a portion of the text of the bulletin there or something.

Also a lower priority nice to have would be able to enable admin-only bulletin previews.

https://www.metaculus.com/admin/misc/bulletin/

**Comment by lsabor** [comment]
Closing as all of these except number 9 and 13, which are now their own issues
9 is now #3025
13 is now #3026
